### PR TITLE
Enable goreleaser

### PR DESCRIPTION
**Pull request by wojciecholszewski-form3**
PR enables creating Github releases by https://goreleaser.com/ tool

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ builds:
       goarch: '386'
     - goos: windows
       goarch: arm
-  binary: '{{ .ProjectName }}_v{{ .Version }}'
+  binary: '{{ .ProjectName }}_v{{ .Version }}_{{ .Os }}_{{ .Arch }}'
```
Comment (pzmi-f3) [inline .goreleaser.yml:33]: Why do we need to change a binary name? I'm not sure that's a correct name for a terraform plugin.

File: Makefile [hunk 2]
```diff
@@ -30,9 +30,13 @@ release:
 	GOOS=windows GOARCH=386 go build -o ./bin/${BINARY}_${VERSION}_windows_386
 	GOOS=windows GOARCH=amd64 go build -o ./bin/${BINARY}_${VERSION}_windows_amd64
 
-.PHONY: gorelease
-gorelease: deps
-	goreleaser
+.PHONY: goreleaser-build
+goreleaser-build: deps
+	goreleaser build --parallelism=1 --snapshot --debug --rm-dist
```
Comment (pzmi-f3) [inline Makefile:35]: Why limit parallelism?

File: Makefile [hunk 3]
```diff
@@ -30,9 +30,13 @@ release:
...
+
+.PHONY: goreleaser-release
+goreleaser-release: deps
+	goreleaser release --parallelism=1 --debug
```
Comment (pzmi-f3) [inline Makefile:39]: ditto

**Review by pzmi-f3 — COMMENTED** [review]
I don't fully understand all change. Could you reply to my comments, please?

File: .goreleaser.yml (hunk 1 above)
Comment (wojciecholszewski-form3) [reply .goreleaser.yml:33]: I thought we should align the names to what we have in the Makefile target which builds binaries by hand.

https://github.com/form3tech-oss/terraform-provider-cortex/blob/9a17bc1878ca2d0d0444d770478f39186b6b108a/Makefile#L20-L31

Happy to revert if it is wrong, shall I?

Comment (pzmi-f3) [reply .goreleaser.yml:33]: I though so too but this Make target is not used anywhere. Upstream project uses GitHub Actions with goreleaser to build binaries. https://github.com/inuits/terraform-provider-cortex/blob/main/.github/workflows/release.yml

File: Makefile (hunk 2 above)
Comment (wojciecholszewski-form3) [reply Makefile]: Discussed f2f and reverted https://github.com/form3tech-oss/terraform-provider-cortex/pull/4/commits/abec800f70df416270298e07143b8d5c720117e6

File: Makefile (hunk 3 above)
Comment (wojciecholszewski-form3) [reply Makefile]: Discussed f2f and reverted https://github.com/form3tech-oss/terraform-provider-cortex/pull/4/commits/abec800f70df416270298e07143b8d5c720117e6

File: .goreleaser.yml (hunk 1 above)
Comment (wojciecholszewski-form3) [reply .goreleaser.yml]: Addressed here https://github.com/form3tech-oss/terraform-provider-cortex/pull/4/commits/22d1c2ef8324d36d61fbc25ab1ee9c6789ec2379

**Review by pzmi-f3 — APPROVED** [review]

**Pull request merged**